test/testapi/integration.py

```diff
@@ -0,0 +1,224 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import logging
+from pathlib import Path
+
+import pytest
+
+from provisioner.vendor import attr
+
+from test import helper as h
+
+from . import defs
+from . import helper
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s
+class HostMeta:
+    # TODO validators for all
+    remote = attr.ib()
+    host = attr.ib()
+    ssh_config = attr.ib()
+    request = attr.ib()
+
+    label = attr.ib(converter=lambda v: '' if not v else v, default='')
+    machine_name = attr.ib(default=None)
+    hostname = attr.ib(default=None)
+    interface = attr.ib(default=None)
+    host_user = attr.ib(default=None)
+
+    _hostname = attr.ib(init=False, default=None)
+    _tmpdir = attr.ib(init=False, default=None)
+    _repo = attr.ib(init=False, default=None)
+    _local_user_home = attr.ib(init=False, default=None)
+    _api_installed = attr.ib(init=False, default=False)
+    _rpm_prvsnr = attr.ib(init=False, default=None)
+    _rpm_prvsnr_cli = attr.ib(init=False, default=None)
+    _rpm_prvsnr_api = attr.ib(init=False, default=None)
+
+    def __attrs_post_init__(self):
+        # TODO more smarter logic to get interface that is asseccible from host
+        # (relates to https://github.com/hashicorp/vagrant/issues/2779)
+        if self.interface is None:
+            if (
+                isinstance(self.remote, h.VagrantMachine) and
+                (self.remote.provider == 'vbox')
+            ):
+                self.interface = 'enp0s8'
```

Comment:
Do I understand correctly that `enp0s8` is the constant interface name for IPv4 settings that are needed for VirtualBox configuration?

Reply:
yes